tests/unit/amo/components/TestUserProfile.js

```diff
@@ -0,0 +1,318 @@
+
+    const root = renderUserProfile({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('dispatches fetchUserAccount action if username param changes', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'killmonger' } });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const { store } = dispatchSignInActions({ username: 'black-panther' });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfile({
+      params: { username: 'black-panther' },
+      store,
+    });
+
+    dispatchSpy.reset();
+
+    sinon.assert.notCalled(dispatchSpy);
+
+    root.setProps({ params: { username: 'black-panther' } });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders the user avatar', () => {
+    const root = renderUserProfile();
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find(UserAvatar)).toHaveProp(
+      'user', getCurrentUser(root.instance().props.store.getState().users));
+  });
+
+  it('still passes user prop to avatar while loading', () => {
+    const root = renderUserProfile({ params: { username: 'not-ready' } });
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find(UserAvatar)).toHaveProp('user', undefined);
+  });
+
+  it("renders the user's name", () => {
+    const root = renderUserProfile();
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find('.UserProfile-name')).toHaveText('Matt MacTofu');
+  });
+
+  it('renders LoadingText when user name is not ready', () => {
+    const root = renderUserProfile({ params: { username: 'not-ready' } });
+
+    const headerContent = root.find('.UserProfile-user-info').prop('header');
+    const header = shallow(<div>{headerContent}</div>);
+
+    expect(header.find('.UserProfile-name').find(LoadingText))
+      .toHaveLength(1);
+  });
+
+  it("renders the user's username if no display name exists", () => {
+    const { store } = dispatchSignInActions({
+      userProps: {
+        display_name: null,
+        userId: 500,
```

Comment:
This doesn't seem relevant to the test

Reply:
It's not, shoot, I remember why I did it and that's no longer relevant. Thanks! 🔥